### PR TITLE
fix for running nss wrapper on alpine

### PR DESCRIPTION
**Pull request by ajixuan**

### CODE REVIEW

File: shtdlib.sh [hunk 1]
```diff
@@ -796,20 +796,39 @@ function init_nss_wrapper {
...
     passwd_string="${TMP_USER}:x:${BUID}:${BGID}:Bob the builder:${TMP_HOME_PATH}:/bin/false"
     group_string="${TMP_GROUP}:x:${BUID}:"
-    passwd_pattern=".*:x:${BUID}:.*:.*:.*:.*"
-    group_pattern=".*:x:${BGID}:.*"
+    passwd_name_pattern='^'"${TMP_USER}"':x:.*:.*:.*:.*:.*$'
+    passwd_pattern='^.*:x:'"${BUID}"':.*:.*:.*:.*$'
+    group_name_pattern="^${TMP_GROUP}"':x:.*:.*$'
+    group_pattern='^.*:x:'"${BGID}"':.*$'
+
+    if grep -qE "${passwd_name_pattern}" "${tmp_passwd_file}"; then
```
Comment (antevens) [inline shtdlib.sh:804]: Suggest using long arguments when scripting

File: shtdlib.sh [hunk 2]
```diff
@@ -796,20 +796,39 @@ function init_nss_wrapper {
...
+      sed -i "s|${passwd_pattern}||g" "${tmp_passwd_file}"
```
Comment (antevens) [inline shtdlib.sh:805]: Suggesting perhaps merging this block with the code that does the group editing (it's the same pattern, just a variable difference).

File: shtdlib.sh (hunk 1 above)
Comment (ajixuan) [reply shtdlib.sh]: long option for -E does not exist in alpine linux

File: shtdlib.sh [hunk 3]
```diff
@@ -801,14 +801,14 @@ function init_nss_wrapper {
     group_name_pattern="^${TMP_GROUP}"':x:.*:.*$'
     group_pattern='^.*:x:'"${BGID}"':.*$'
 
-    if grep --extended-regexp --quiet "${passwd_name_pattern}" "${tmp_passwd_file}" \
```
Comment (antevens) [inline shtdlib.sh]: Any reason why you want to use short options rather than the long, are the long not supported on Alpine?

Comment (ajixuan) [reply shtdlib.sh]: Yeah thats right, this was causing error on the postgres container

Comment (antevens) [reply shtdlib.sh]: That's annoying

**Review by antevens — APPROVED** [review]
It bugs me a bit that we can't use long options, I'm tempted to do an OS check but that's even more annoying but then again we don't want to convert all our functions to the short options since they would be so much less readable and maintainable.

**Pull request merged**